docs/plugins/icons.js

```diff
@@ -1,50 +1,44 @@
 const { Icons, Spots } = require("@stackoverflow/stacks-icons");
+const fs = require("fs/promises");
+
+function modifySvg(content, type, name, classes, dimension) {
+  var defaultClasses = `svg-${type} ${type}${name}`;
+
+  if (!content) {
+    return `<span class="fc-danger">Invalid ${type}: ${name}</span>`;
+  }
+
+  // If we have classes, add them
+  if (classes != null) {
+    content = content.replace(defaultClasses, defaultClasses + " " + classes);
+  }
+
+  // If we need to change the size, do that too
+  if (dimension != null) {
+    content = content.replace('width="18" height="18"', 'width="' + dimension + '" height="' + dimension + '"');
+  }
+
+  return content;
+}
 
 module.exports = {
   configFunction(eleventyConfig) {
     // Icon shortcode
     eleventyConfig.addLiquidShortcode("icon", function(name, classes, dimension) {
       var svg = Icons["Icon" + name];
-      var defaultClasses = "svg-icon icon" + name;
-
-      if (!svg) {
-        return `<span class="fc-danger">Invalid icon: ${name}</span>`;
-      }
-
-      // If we have classes, add them
-      if (classes != null) {
-        svg = svg.replace(defaultClasses, defaultClasses + " " + classes);
-      }
-
-      // If we need to change the size, do that too
-      if (dimension != null) {
-        svg = svg.replace('width="18" height="18"', 'width="' + dimension + '" height="' + dimension + '"');
-      }
-
-      return svg;
+      return modifySvg(svg, "icon", name, classes, dimension);
     });
 
     // Spot shortcode
     eleventyConfig.addLiquidShortcode("spot", function(name, classes, dimension) {
       var svg = Spots["Spot" + name];
+      return modifySvg(svg, "spot", name, classes, dimension);
+    });
 
-      if (!svg) {
-        return `<span class="fc-danger">Invalid spot: ${name}</span>`;
-      }
-
-      var defaultClasses = "svg-spot spot" + name;
-
-      // If we have classes, add them
-      if (classes != null) {
-        svg = svg.replace(defaultClasses, defaultClasses + " " + classes);
-      }
-
-      // If we need to change the size, do that too
-      if (dimension != null) {
-        svg = svg.replace('width="18" height="18"', 'width="' + dimension + '" height="' + dimension + '"');
-      }
-
-      return svg;
+    // embed svgs in the assets folder
+    eleventyConfig.addLiquidShortcode("embed", async function(name, classes) {
+      const file = await fs.readFile(`assets/img/icons/${name}.svg`, "utf8");
```

Comment:
nit: I wonder if we should simply convert all the `var` in this file into `const` and `let` since we are at it. It probably helps with consistency in the long term. :)

Reply:
If we add in the `@stackoverflow/eslint-config` package, then it'll start enforcing lots of best practices. I'm going to leave that change for a rainy day (follow-up PR some time in the future).